### PR TITLE
unit tests for Channel Info ViewModel, Channels ViewModel, Channels Members ViewModel, and Channel Notification View Model 

### PR DESCRIPTION
**Pull request by Cdf-creator**
This commit contains the UNIT TESTS I did on the Channel Info ViewModel, Channels ViewModel, Channels Members ViewModel, Channel Notification View Model and Channel Page ViewModel.

### CODE REVIEW

File: lib/ui/view/channel/channel_members/channel_members_list_model.dart [hunk 1]
```diff
@@ -67,6 +67,6 @@ class ChannelMembersModel extends BaseViewModel {
   }
 
   void goBack() {
-    NavigationService().back();
+    navigator.back();
```
Comment (pauleke65) [inline lib/ui/view/channel/channel_members/channel_members_list_model.dart:70]: rename to navigation service

File: test/view_models/channel_notification_viewmodel_test.dart [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+// This is a test setup
+void main() {
+  group('ChannelNotificationViewModelTest -', () {
+    setUp(() => registerServices());
+    tearDown(() => unregisterServices());
+    group('toggleOptions -', () {
+      test(
+          'When toggleOptions is called and a string value is entered, should be true.',
+          () {
```
Comment (pauleke65) [inline test/view_models/channel_notification_viewmodel_test.dart:13]: No test here!

**Review by pauleke65 — CHANGES_REQUESTED** [review]
Changes requested

File: test/view_models/channels_viewmodel_test.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hng/models/app_model.dart';
+import 'package:hng/models/channel_members.dart';
+import 'package:hng/ui/view/channel/channel_list/channels_viewmodel.dart';
+import '../helpers/test_helpers.dart';
+
+// This is a test setup
+void main() {
```
Comment (pauleke65) [inline test/view_models/channels_viewmodel_test.dart:8]: Please write the right tests for the channels view model not the CHANNELS LIST view model

Comment (Cdf-creator) [reply test/view_models/channels_viewmodel_test.dart:8]: Check the name of the folder, it is channel view model. The person that writes the logic for the channels view model wrote it as "CHANNELS LIST view model".

Comment (Cdf-creator) [reply test/view_models/channels_viewmodel_test.dart:8]: I wrote unit test for the right file which is "channels view model"

File: lib/ui/view/channel/channel_members/channel_members_list_model.dart (hunk 1 above)
Comment (Cdf-creator) [reply lib/ui/view/channel/channel_members/channel_members_list_model.dart:70]: It is still the same thing.

navigator = locator<NavigationService>();
    It's still navigation service that was used.
    The test kills returning a null value if NavigationService is used directly inside of navigator.

Comment (pauleke65) [reply lib/ui/view/channel/channel_members/channel_members_list_model.dart:70]: I never meant NavigationService itself.

I meant change the variable name to ```_navigationService``` for uniformity

**Comment by YenumJay** [comment]
Please fix issues from Analyzer

**Comment by YenumJay** [comment]
resolve conflicts

**Review by pauleke65 — APPROVED** [review]

**Review by jeremiahseun — APPROVED** [review]

**Pull request merged**